src/utils/createCoreModule/index.ts

```diff
@@ -0,0 +1,175 @@
+
+/** Different types of modules */
+export type CoreModuleName
+  = 'farmer'
+  | 'milk'
+  | 'delivery'
+  | 'loan'
+  ;
+
+/*--------------------------------- Functions ---------------------------------*/
+
+export function createMutations<Row>(): MutationHandlers<CoreModuleState<Row>> {
+  return {
+    setRows (state: CoreModuleState<Row>, { rows }: SetPayload<Row>) {
+      state.rows = rows;
+    },
+
+    createRow (state: CoreModuleState<Row>, { row }: CreatePayload<Row>) {
+      state.rows.push(row);
+    },
+
+    updateRow (state: CoreModuleState<Row>, { row }: UpdatePayload<Row>) {
+      state.rows = updateElementByProp('uuid', row, state.rows);
+    },
+  };
+}
+
+function createActions<Row>(path: CorePath): ActionHandlers<CoreModuleState<Row>> {
+  return {
+    async createRow ({ commit }, { row }: CreatePayload<Row>) {
+
+      const payload: CreationMutateRow<Row> = {
+        ...(row as any),
+        lastModified: new Date().toUTCString(),
+        uuid: uuid4(),
+        status: 'local',
+      };
+      commit('createRow', { row: payload });
+
+      const { uuid } = payload;
+
+
+      // Send the new row to the core
+      const api = new CoreAPI(path);
+
+      // Create new block for block scoped variables (let) to avoid errors
+      { let lastModified: string;
+
+        // Attempt to create resource on core
+        try {
+          ({ lastModified } = await api.create(payload));
+        } catch (err) {
+          // Failed to create resource on Core
+          if (isResponse(err)) {
+            const response = err;
+            // tslint:disable-next-line:no-console
+            console.log(response.status);
+
+            return uuid;
+          } else {
+            // Not a response error, should be logged
+            // tslint:disable-next-line:no-console
+            console.log(err.message || err);
+            return uuid;
+          }
+        }
+
+        // Create updated model and update store
+        commit('updateRow', {
+          row: {
+            uuid,
+            lastModified,
+            status: 'clean',
+          },
+        });
+        return uuid;
+      }
+    },
+
+    async updateRow ({ commit, state }, { row }: UpdatePayload<Row>) {
+      if (!hasUUID(row)) {
+        throw new Error('Update must contain UUID');
+      }
+      const { uuid } = row;
+      const payload = {
+        ...(row as any),
+        lastModified: new Date().toUTCString(),
+        status: 'modified',
+      } as UpdateMutateRow<Row>;
+
+      // Update the row in store
+      commit('updateRow', { row: payload });
+
+      // Send update to the core
+      const api = new CoreAPI(path);
+
+      { let coreRow: CoreRow<Row>;
+
+        try {
+          ({ row: coreRow } = await api.update<any>(payload));
+        } catch (err) {
+          // Failed to create resource on Core
+          if (isResponse(err)) {
+            // TODO: Deal with different core errors
```

Comment:
still need to do this?

Reply:
More of a note for people that come after us if they even end up using this